test/sql-tap/gh-6024-funcs-return-bin.c

```diff
@@ -0,0 +1,41 @@
+#include "msgpuck.h"
+#include "module.h"
+
+enum {
+	BUF_SIZE = 512,
+};
+
+int
+ret_bin(box_function_ctx_t *ctx, const char *args, const char *args_end)
+{
+	uint32_t arg_count = mp_decode_array(&args);
+	if (arg_count != 1) {
+		return box_error_set(__FILE__, __LINE__, ER_PROC_C,
+				     "invalid argument count");
+	}
+	if (mp_typeof(*args) != MP_BIN) {
+		return box_error_set(__FILE__, __LINE__, ER_PROC_C,
+				     "argument should be varbinary");
+	}
+	const char* buf;
+	uint32_t buf_size;
+	buf = mp_decode_bin(&args, &buf_size);
```

Comment:
Actually you don't have to decode argument since an issue is about _returning_ varbinary, so you shrink a bit this function - just encode as binary any string. Up to you, it won't obviously affect semantic of the test.

Reply:
Thank you for the review! Fixed, removed unnecessary parts of the test.